jwst/scripts/world_coords.py

```diff
@@ -258,129 +272,125 @@ def get_update_method(methods, model, mode):
-        imhdu.header['SLIT'] = 'SLIT_{0}'.format(i)
+        imhdu.header["PLANE1"] = "lambda, microns"
+        imhdu.header["PLANE2"] = f"{output_frame}_x, arcsec"
+        imhdu.header["PLANE3"] = f"{output_frame}_y, arcsec"
+        imhdu.header["PLANE4"] = "slit_y, relative to center (0, 0)"
+        imhdu.header["SLIT"] = f"SLIT_{i}"
 
         # -1 and +1 are to express this in 1-based coordinates
-        imhdu.header['CRVAL1'] = (
+        imhdu.header["CRVAL1"] = (
             model.meta.subarray.xstart - 1 + int(_toindex(slit.bounding_box[0][0])) + 1
         )
-        imhdu.header['CRVAL2'] = (
+        imhdu.header["CRVAL2"] = (
             model.meta.subarray.xstart - 1 + int(_toindex(slit.bounding_box[1][0])) + 1
         )
 
         # Input coordinates will be 1-based.
-        imhdu.header['CRPIX1'] = 1
-        imhdu.header['CRPIX2'] = 1
-        imhdu.header['CTYPE1'] = 'pixel'
-        imhdu.header['CTYPE2'] = 'pixel'
+        imhdu.header["CRPIX1"] = 1
+        imhdu.header["CRPIX2"] = 1
+        imhdu.header["CTYPE1"] = "pixel"
+        imhdu.header["CTYPE2"] = "pixel"
         hdulist.append(imhdu)
     return hdulist
 
 
 def imaging_coords(model):
     """
-    Computes wavelengths, and space coordinates of a NIRSPEC
-    imaging exposure after running assign_wcs.
+    Compute wavelengths and space coordinates of a NIRSPEC imaging exposure after assign_wcs.
 
     Parameters
     ----------
     model : ImageModel
         An image with extracted slits, i.e. the output of extract2d.
 
-
-    returns a fits hdulist
+    Returns
+    -------
+    hdulist : fits object
+        Fits hdulist
     """
     hdulist = fits.HDUList()
     phdu = fits.PrimaryHDU()
-    phdu.header['filename'] = model.meta.filename
-    phdu.header['data'] = 'world coordinates'
+    phdu.header["filename"] = model.meta.filename
+    phdu.header["data"] = "world coordinates"
     hdulist.append(phdu)
 
     output_frame = model.available_frames[-1]
-    bb = model.meta.wcs.bounding_box
-    x, y = wcstools.grid_from_bounding_box(bb, step=(1, 1), center=True)
-    ra, dec, lam = slit(x + 1, y + 1)  # noqa
-    world_coordinates = np.array([lam, ra, dec])
-
-    imhdu = fits.ImageHDU(data=world_coordinates)
-    imhdu.header['PLANE1'] = 'lambda, microns'
-    imhdu.header['PLANE2'] = '{0}_x, deg'.format(output_frame)
-    imhdu.header['PLANE3'] = '{0}_y, deg'.format(output_frame)
-    imhdu.header['SLIT'] = 'SLIT_{0}'.format(i)  # noqa
-
-    # add the overall subarray offset
-    imhdu.header['CRVAL1'] = model.meta.subarray.xstart - 1 + int(_toindex(bb[0][0]))
-    imhdu.header['CRVAL2'] = model.meta.subarray.ystart - 1 + int(_toindex(bb[1][0]))
-
-    # Input coordinates will be 1-based.
-    imhdu.header['CRPIX1'] = 1
-    imhdu.header['CRPIX2'] = 1
-    imhdu.header['CTYPE1'] = 'pixel'
-    imhdu.header['CTYPE2'] = 'pixel'
-    hdulist.append(imhdu)
+    for i, slit in enumerate(output_frame):
```

Comment:
Why was this for loop added?

Reply:
Because "slit" and "i" were not defined and the code was breaking.